src/read.c

```diff
@@ -1248,15 +1267,16 @@ void ReadFuncExpr (
         Match( S_SEMICOLON, ";", STATBEGIN|S_END|follow );
     }
 
-    /* 'function( ... arg )' takes a variable number of arguments              */
-    if (narg >= 1 && ! strcmp( "arg", CSTR_STRING( ELM_LIST(nams, narg) ) ) )
+    /* 'function( a,b, ...p )' takes a variable number of arguments           */
+    /* Also, we special case function(arg)                                   */
+    if (isvarg || ( narg >= 1 && ! strcmp( "arg", CSTR_STRING( ELM_LIST(nams, narg) ) )) )
```

Comment:
I guess the `narg >= 1` should be changed to `narg == 1` here, so that only the old `function(arg)` leads to a variadic function, while `function(a,b,c,arg)` is (again, as in the past) a simple 4 parameter function.